pkg/app/piped/executor/kubernetes/sync_test.go

```diff
@@ -19,17 +19,166 @@ import (
+						}, &unstructured.Unstructured{}),
+					}, nil)
+					return p
+				}(),
+				config: &config.KubernetesDeploymentSpec{
+					GenericDeploymentSpec: config.GenericDeploymentSpec{
+						Pipeline: &config.DeploymentPipeline{},
+					},
+				},
+			},
+		},
+		{
+			name: "unable to apply manifests",
+			want: model.StageStatus_STAGE_FAILURE,
+			executor: &Executor{
+				Input: executor.Input{
+					Deployment: &model.Deployment{
+						Trigger: &model.DeploymentTrigger{
+							Commit: &model.Commit{},
+						},
+					},
+					PipedConfig:  &config.PipedSpec{},
+					LogPersister: &fakeLogPersister{},
+					AppManifestsCache: func() cache.Cache {
+						c := cachetest.NewMockCache(ctrl)
+						c.EXPECT().Get(gomock.Any()).Return(nil, fmt.Errorf("not found"))
+						c.EXPECT().Put(gomock.Any(), gomock.Any()).Return(nil)
+						return c
+					}(),
+					Logger: zap.NewNop(),
+				},
+				provider: func() provider.Provider {
+					p := providertest.NewMockProvider(ctrl)
+					p.EXPECT().LoadManifests(gomock.Any()).Return([]provider.Manifest{
+						provider.MakeManifest(provider.ResourceKey{
+							APIVersion: "apps/v1",
+							Kind:       provider.KindDeployment,
+						}, &unstructured.Unstructured{
+							Object: map[string]interface{}{"spec": map[string]interface{}{}},
+						}),
+					}, nil)
+					p.EXPECT().ApplyManifest(gomock.Any(), gomock.Any()).Return(fmt.Errorf("error"))
+					return p
+				}(),
+				config: &config.KubernetesDeploymentSpec{
+					QuickSync: config.K8sSyncStageOptions{
+						AddVariantLabelToSelector: true,
+					},
+				},
+			},
+		},
+		{
+			name: "successfully apply manifests",
+			want: model.StageStatus_STAGE_SUCCESS,
+			executor: &Executor{
+				Input: executor.Input{
+					Deployment: &model.Deployment{
+						Trigger: &model.DeploymentTrigger{
+							Commit: &model.Commit{},
+						},
+					},
+					PipedConfig:  &config.PipedSpec{},
+					LogPersister: &fakeLogPersister{},
+					AppManifestsCache: func() cache.Cache {
+						c := cachetest.NewMockCache(ctrl)
+						c.EXPECT().Get(gomock.Any()).Return(nil, fmt.Errorf("not found"))
+						c.EXPECT().Put(gomock.Any(), gomock.Any()).Return(nil)
+						return c
+					}(),
+					Logger: zap.NewNop(),
+				},
+				provider: func() provider.Provider {
+					p := providertest.NewMockProvider(ctrl)
+					p.EXPECT().LoadManifests(gomock.Any()).Return([]provider.Manifest{
+						provider.MakeManifest(provider.ResourceKey{
+							APIVersion: "apps/v1",
+							Kind:       provider.KindDeployment,
+						}, &unstructured.Unstructured{
+							Object: map[string]interface{}{"spec": map[string]interface{}{}},
+						}),
+					}, nil)
+					p.EXPECT().ApplyManifest(gomock.Any(), gomock.Any()).Return(nil)
+					return p
+				}(),
+				config: &config.KubernetesDeploymentSpec{
+					QuickSync: config.K8sSyncStageOptions{
+						AddVariantLabelToSelector: true,
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
```

Comment:
nit: `tt` -> `tc`, `tests` -> `testcases`

Reply:
Right, hopefully the code genereted by Jetbrains Goland can configurable...